### PR TITLE
Added a new observer type 'pitr'

### PR DESCRIPTION
**Pull request by YuJuncen**
This PR added a new observer handle type `pitr` for resolving the problem of conflicting observer type with CDC.

### CODE REVIEW

File: components/raftstore/src/coprocessor/mod.rs [hunk 1]
```diff
@@ -247,11 +246,20 @@ impl ObserveHandle {
 pub struct CmdObserveInfo {
     pub cdc_id: ObserveHandle,
     pub rts_id: ObserveHandle,
+    pub pitr_id: ObserveHandle,
 }
```
Comment (overvenus) [inline components/raftstore/src/coprocessor/mod.rs:250]: What about an enum? Also we can replace `ObserverType` with `CmdObserveInfo`.

Comment (YuJuncen) [reply components/raftstore/src/coprocessor/mod.rs:250]: It seems that `CmdObserverInfo` may contain multi types of observers (e.g. when the cluster enabled both `resolved-ts` and `cdc`, both of `cdc_id` and `rts_id` would be non-default.) So I'm afraid we cannot make it an `enum` trivially. 🤔

File: components/backup-stream/src/event_loader.rs [hunk 2]
```diff
@@ -346,7 +346,7 @@ where
                     .register_region(&r.region, handle.clone(), Some(start_ts));
                 let region_id = r.region.get_id();
                 let snap = self.observe_over_with_retry(&r.region, move || {
-                    ChangeObserver::from_cdc(region_id, handle.clone())
+                    ChangeObserver::from_pitr(region_id, handle.clone())
```
Comment (3pointer) [inline components/backup-stream/src/event_loader.rs:349]: should it be some components just like `backup_log` `bklog`?

**Review by 3pointer — APPROVED** [review]
LGTM

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**